### PR TITLE
VIH-6758 joh waiting room

### PR DESCRIPTION
**Pull request by shaed-parkar**
### JIRA link (if applicable) ###

VIH-6758
VIH-6840

### Change description ###

* New JOH waiting room
* Refactor waiting room and their spec files to minimise duplication
* Map JOH user roles and use user role instead of hearing role

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/joh-waiting-room/joh-waiting-room.component.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            userMediaService,
+            userMediaStreamService,
+            notificationSoundsService
+        );
+    }
+
+    ngOnInit(): void {
+        this.audioOnly = false;
+        this.errorCount = 0;
+        this.logger.debug(`${this.loggerPrefixJOH} Loading participant waiting room`);
```
Comment (levibutler) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/joh-waiting-room/joh-waiting-room.component.ts:73]: joh

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/participant-waiting-room.component.ts [hunk 2]
```diff
@@ -113,8 +112,8 @@ export class ParticipantWaitingRoomComponent extends WaitingRoomBaseComponent im
         }
     }
 
-    announceHearingIsAboutToStart(): void {
-        this.notificationSoundsService.playHearingAlertSound();
+    async announceHearingIsAboutToStart(): Promise<void> {
```
Comment (levibutler) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/participant-waiting-room.component.ts:115]: i cant comment on code you havnt changed yet :P
could we use executeWaitingRoomCleanup ?

Comment (levibutler) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/participant-waiting-room.component.ts:115]: in ngOnDestroy

Comment (shaed-parkar) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/participant-waiting-room.component.ts:115]: we can! done!

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/joh-waiting-room/joh-waiting-room.component.ts (hunk 1 above)
Comment (shaed-parkar) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/joh-waiting-room/joh-waiting-room.component.ts:73]: done

**Comment by levibutler** [comment]
Given the upcoming changes to rooms (virtual/ WR/conference) refactoring now may be premature optimization and require more work to unpick. For this reason, I think the flagged code duplication is acceptable at this time.

**Review by levibutler — APPROVED** [review]

**Pull request merged**